docs/developer-setup.md

```diff
@@ -0,0 +1,170 @@
+# Development Workflow
+
+## Prerequisites
+
+* You have Go 1.9 installed on your local host/development machine.
+* You have Docker installed on your local host/development machine. Docker is required for building NDM container images and to push them into a Kubernetes cluster for testing. 
+* You have `kubectl` installed. For running integration tests, you will either require to use an existing cluster or NDM will create a Minikube cluster on the local host/development machine. Don't worry if you don't have access to the Kubernetes cluster, raising a PR with the NDM repository will run integration tests for your changes against a Minikube cluster.
+
+## Initial Setup
+
+### Fork in the cloud
+
+1. Visit https://github.com/openebs/node-disk-manager
+2. Click `Fork` button (top right) to establish a cloud-based fork.
+
+### Clone fork to local host
+
+Place openebs/node-disk-manager's code on your `GOPATH` using the following cloning procedure.
+Create your clone:
+
+```sh
+
+mkdir -p $GOPATH/src/github.com/openebs
```

Comment:
This and several following commands will fail, if `GOPATH` has more than one (`:` separated) folders in its path.

Sorry, to giving comments after merge. :disappointed: